osu.Server.Spectator/Hubs/ScoreUploader.cs

```diff
@@ -33,22 +28,29 @@ public double UploadInterval
         private readonly ConcurrentQueue<UploadItem> queue = new ConcurrentQueue<UploadItem>();
         private readonly IDatabaseFactory databaseFactory;
         private readonly IScoreStorage scoreStorage;
-        private readonly Timer timer;
-        private readonly CancellationTokenSource timerCancellationSource;
-        private readonly CancellationToken timerCancellationToken;
+        private readonly CancellationTokenSource cancellationSource;
+        private readonly CancellationToken cancellationToken;
 
         public ScoreUploader(IDatabaseFactory databaseFactory, IScoreStorage scoreStorage)
         {
             this.databaseFactory = databaseFactory;
             this.scoreStorage = scoreStorage;
 
-            timerCancellationSource = new CancellationTokenSource();
-            timerCancellationToken = timerCancellationSource.Token;
+            cancellationSource = new CancellationTokenSource();
+            cancellationToken = cancellationSource.Token;
 
-            timer = new Timer(50);
-            timer.AutoReset = false;
-            timer.Elapsed += (_, _) => Task.Run(Flush);
-            timer.Start();
+            Task.Run(runFlushLoop);
```

Comment:
Am I reading this correct, that this is infinite-looping a TPL thread? This can't be good. At the very least this should be a long running thread.

Reply:
Oh yep, for sure.